src/main/java/bio/terra/service/snapshot/Snapshot.java

```diff
@@ -180,4 +181,13 @@ public AzureRegion getStorageAccountRegion() {
   public boolean isSecureMonitoringEnabled() {
     return getFirstSnapshotSource().getDataset().isSecureMonitoringEnabled();
   }
+
+  public Map<String, Object> getMode() {
```

Comment:
I think you could probably have this be slightly more strongly typed by having mode be a `SnapshotRequestContentsModel`

Reply:
Also because we want this to be more than mode: we want to have it include all of the things (like what query was used)

Reply:
I do store the full definition, for example, with a byQuery
I do mode.put("querySpec", queryModel);
and then use jackson to write the object as json, which results in
{ mode: byQuery, querySpec: { assetName: asset, query: “SELECT * etc..." } }

Reply:
I do store the full definition, for example, with a byQuery
I do `mode.put("querySpec", queryModel);
`and then use jackson to write the object as json, which results in
`{ mode: byQuery, querySpec: { assetName: asset, query: “SELECT * etc..." } }
`
But you make a good point that I can just use the entire `SnapshotRequestContentsModel`
it does result in visible null values though:
`{"mode": "byFullView", "assetSpec": null, "querySpec": null, "rowIdSpec": null, "datasetName": "TestDatasetCombined"}`